### PR TITLE
Iterate over each page when retrieving OCS OLM Operator tags

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #3950 

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/utils.py [hunk 1]
```diff
@@ -1969,17 +1970,28 @@ def get_ocs_olm_operator_tags(limit=100):
...
+        )
+        if not resp.ok:
+            raise requests.RequestException(resp.json())
+        tags = resp.json()["tags"]
+        if len(tags) == 0:
+            log.info("No more tags to retrieve")
+            break
+        log.debug(tags)
+        all_tags.append(tags)
+        page += 1
```
Comment (vasukulkarni) [inline ocs_ci/utility/utils.py:1993]: shouldn't the while look use some kind of page <= max? how is the next loop using incremented page?

Comment (clacroix12) [reply ocs_ci/utility/utils.py:1993]: I tested this locally and the API gracefully accepted an "empty page" and just returned an empty tag list which is what I used to determine that we reached the end.

Comment (clacroix12) [reply ocs_ci/utility/utils.py:1993]: From my local test output:
```
Retrieving OCS OLM Operator tags (limit 100, page 8)
[]
No more tags to retrieve
```

**Review by vasukulkarni — APPROVED** [review]

**Comment by clacroix12** [comment]
> PR validation
> 
> Cluster Name:
> Cluster Configuration:
> PR Test Suite: deployment
> PR Test Path: tests/
> Additional Test Params:
> OCP VERSION: 4.5
> OCS VERSION: 4.5
> 
> [Job FAILED](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/279//testReport) (installation failed, tests not executed).

I was using `append` instead of `extend` to build out the list of tags so it was actually returning a list of lists. I fixed this and will re-run the test job.

**Comment by clacroix12** [comment]
```
01:10:12 - MainThread - ocs_ci.deployment.deployment - INFO - Adding CatalogSource
01:10:12 - MainThread - ocs_ci.utility.utils - INFO - Retrieving the authentication config dictionary
01:10:12 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 1)
01:10:13 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 2)
01:10:13 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 3)
01:10:14 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 4)
01:10:14 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 5)
01:10:15 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 6)
01:10:16 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 7)
01:10:16 - MainThread - ocs_ci.utility.utils - INFO - Retrieving OCS OLM Operator tags (limit 100, page 8)
01:10:17 - MainThread - ocs_ci.utility.utils - INFO - No more tags to retrieve
01:10:17 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  labels:
    ocs-operator-internal: 'true'
  name: ocs-catalogsource
  namespace: openshift-marketplace
spec:
  displayName: Openshift Container Storage
  icon:
    base64data: ''
    mediatype: ''
  image: quay.io/rhceph-dev/ocs-registry:4.5.2-146.ci
  publisher: Red Hat
  sourceType: grpc
  ```
  
  Marking as verified

**Review by shylesh — APPROVED** [review]
+1

**Review by petr-balogh — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**